elyra/pipeline/registry.py

```diff
@@ -768,7 +772,10 @@ def get_properties(self, processor_type, component_id):
 
             component_path = component['path'][reader._type]
             if reader._type == "filename":
-                component_path = os.path.join(os.path.dirname(__file__), component_path)
+                component_resource_dir = ""
+                if processor_type in ['kfp', 'airflow']:
+                    component_resource_dir = "resources/" + processor_type
```

Comment:
I'm confused between `processor_type` and `parser._type` and would prefer that we consolidate these if possible.  Since 'local' is not a formal processor type, I suspect it's an invariant condition that this method is ever called when `processor_type == 'local'` - is that a correct statement?

We should also try to avoid the `in` clause, otherwise we'll require code changes on each new processor.

This question seems like one fundamental to the overall design, so I'm not sure it can be addressed here.

Also, I believe this area of the code is being changed in #1801 (and eliminated), so it may be moot altogether.

Reply:
Yeah, you're correct that the way we implement this particular portion of the code will be changed once the refactoring is done. It will remove the need to use `in` and we'll put the onus on the component parser to determine it's own path to the local file, then just pass that path to the reader

Reply:
will wait for the refactoring to complete then to make the changes, thx